wyzesense2mqtt/wyzesense2mqtt.py

```diff
@@ -424,52 +521,118 @@ def on_message_reload(MQTT_CLIENT, userdata, msg):
+            (event_type, sensor_state, sensor_battery, sensor_signal) = event.Data
+            event_payload.update(
+                {'signal_strength': sensor_signal * -1, 'battery': sensor_battery}
+            )
 
             # Add sensor if it doesn't already exist
-            if (event.MAC not in SENSORS):
-                add_sensor_to_config(event.MAC, sensor_type, None)
-                if(CONFIG['hass_discovery']):
+            if event.MAC not in SENSORS:
+                add_sensor_to_config(event.MAC, event_type, None)
+                if CONFIG['hass_discovery']:
                     send_discovery_topics(event.MAC)
 
-            # Build event payload
-            event_payload = {
-                'event': event.Type,
-                'available': True,
-                'mac': event.MAC,
-                'device_class': DEVICE_CLASSES.get(sensor_type),
-                'last_seen': event.Timestamp.timestamp(),
-                'last_seen_iso': event.Timestamp.isoformat(),
-                'signal_strength': sensor_signal * -1,
-                'battery': sensor_battery
-            }
+        if event.Type in ["alarm", "status"]:
+            event_payload['device_class'] = DEVICE_CLASSES.get(event_type)
+            state_topic = f"{CONFIG['self_topic_root']}/{event.MAC}"
 
-            if (CONFIG['publish_sensor_name']):
+            if CONFIG['publish_sensor_name']:
                 event_payload['name'] = SENSORS[event.MAC]['name']
 
             # Set state depending on state string and `invert_state` setting.
             #     State ON ^ NOT Inverted = True
             #     State OFF ^ NOT Inverted = False
             #     State ON ^ Inverted = False
             #     State OFF ^ Inverted = True
-            event_payload['state'] = int((sensor_state in STATES_ON) ^ (SENSORS[event.MAC].get('invert_state')))
+            event_payload['state'] = int(
+                (sensor_state in STATES_ON) ^ (SENSORS[event.MAC].get('invert_state'))
+            )
 
+            mqtt_publish(state_topic, event_payload)
             LOGGER.debug(event_payload)
+        elif event.Type == "keypad":
+            state_payload = {"state": sensor_state}
 
             state_topic = f"{CONFIG['self_topic_root']}/{event.MAC}"
+            event_topic = f"{CONFIG['self_topic_root']}/{event.MAC}/{event_type}"
+            pin_topic = f"{CONFIG['self_topic_root']}/{event.MAC}/pin"
+            set_topic = f"{CONFIG['self_topic_root']}/{event.MAC}/set"
+
             mqtt_publish(state_topic, event_payload)
-        else:
-            LOGGER.debug(f"Non-state event data: {event}")
+
+            if event_type == "mode" and not validate_pin_entry(event.MAC, sensor_state):
+                return
+
+            if event_type in ["mode", "motion"]:
+                mqtt_publish(event_topic, state_payload)
+                mqtt_publish(pin_topic, {'state': False})
```

Comment:
This bit makes sure that the entered PIN is cleared after any action is complete, so that it isn't stored in the MQTT broker.